### ISSUE TITLE
Tags with same key being overwritten

### ISSUE THREAD

**Issue by aaronlippold** [inspec/inspec#2272]
When a control has a tag with the same name on multiple lines inspec is not pushing the data onto a stack but rather just taking the last value. This is causing data loss in the json and other output formats. 

I am hoping we can update the processing to be a little smarter so that when we hit a 'key' that is already defined we could push onto an array or hash so that the data is retained. In the example below my reports would only give back 
```
 tag "cci": "CCI-001814"
 tag "nist": ["CM-5 (1)", "Rev_4"]
```

The user could also do this on their side, i.e. put things into an array or hash and inspec would handle it but I think in terms of user experience, we should also make inspec smart to push this onto an array and perhaps even 'unique' it so that repeated elements in the stack - such as Rev_4' and CM-5 (1) in this example - isn't repeated.

For example, 
```
control "V-72259" do
  title "The SSH daemon must not permit ..."
  desc  "GSSAPI authentication is used ...."
  impact 0.5
  tag "severity": "medium"
  tag "gtitle": "SRG-OS-000364-GPOS-00151"
  tag "gid": "V-72259"
  tag "rid": "SV-86883r2_rule"
  tag "stig_id": "RHEL-07-040430"
  tag "cci": "CCI-000318"
  tag "nist": ["CM-3 f", "Rev_4"]
  tag "cci": "CCI-000368"
  tag "nist": ["CM-6 c", "Rev_4"]
  tag "cci": "CCI-001812"
  tag "nist": ["CM-11 (2)", "Rev_4"]
  tag "cci": "CCI-001813"
  tag "nist": ["CM-5 (1)", "Rev_4"]
  tag "cci": "CCI-001814"
  tag "nist": ["CM-5 (1)", "Rev_4"]
  tag "check": "..."

  describe sshd_config do
    its('GSSAPIAuthentication') { should cmp 'no' }
  end
end
```

**Comment by adamleff** [comment]
I'm :-1: on this feature request. If a user creates the tag first with a value of a Hash and then tries to add a String, what would the experience be? The only sane thing to do is to assume an Array, but I'm very much against assumptions and magic; we need to embrace the element of least surprise.

Instead, I'd rather see us warn the user that they're overwriting the tag so they can go back and fix their control to declare their tag once and only once, and ensure all the values they wish to be included are indeed included, and in the manner they wish them to be stored.

**Comment by miah** [comment]
__What’s happening? Why was this issue closed?__
This issue was closed due to some much needed review of legacy issues or issues that were spawned in older versions of InSpec, i.e. < v3.

__Why do I care?__
You would care about this if this was an issue that you are still seeing and/or feel needs to be addressed in the current version of InSpec.

__What do I need to do?__
If this issue is no longer important, no further action is necessary. However, if you think this is something that should be addressed, please open a new issue and refer to the original issue in the description.
